### DISCUSSION TITLE
We now can have a sidebar-slider without additional plugins

### DISCUSSION THREAD

**Discussion by BurningTreeC** [TiddlyWiki/TiddlyWiki5#5861] [Ideas]
Hello all,

since we have the `<$eventcatcher>` widget, we can now create a "slider" div that allows changing the story-river and sidebar width by dragging. we don't need additional plugins to do so (no need for hammerjs or dynaview)

would this be a welcome addition to the core?

**Comment by BurningTreeC** [comment]
Here's an example of what I'm talking about. All needed tiddlers are open in the story river

[tiddlywiki.zip](https://github.com/Jermolene/TiddlyWiki5/files/6790353/tiddlywiki.zip)

**Reply by BurningTreeC** [reply]
currently it works for fluid-fixed only but it can easily be adopted for fixed-fluid (when I have the time)

**Comment by Jermolene** [comment]
> since we have the `<$eventcatcher>` widget, we can now create a "slider" div that allows changing the story-river and sidebar width by dragging. we don't need additional plugins to do so (no need for hammerjs or dynaview)

Awesome, I should like to see that.

> would this be a welcome addition to the core?

Potentially. I guess what we'd really want for the core would be a generic system for draggable resizing of objects. It would be nice if we could use it not only with the current story layout, but also with the planned flexbox layout (#4473). There might be other places it would be useful too, like resizing modals.

**Reply by BurningTreeC** [reply]
Hi @Jermolene, I've already built such a slider which works with the flexbox layout. It changes the flex-grow for two flex-box "areas" accordingly ... so it can be done ... it would just involve some filtering in the css stylesheet for the layout that's used

**Comment by BurningTreeC** [comment]
Here's an actualized version which works for fluid-fixed and fixed-fluid layouts alike

[tiddlywiki.zip](https://github.com/Jermolene/TiddlyWiki5/files/6792449/tiddlywiki.zip)

**Reply by clutterstack** [reply]
@BurningTreeC I like the idea and the progress looks good. I guess not everything is getting resized in the fixed-fluid mode yet, though.

![fixed-fluid-resize](https://user-images.githubusercontent.com/5599856/125124171-af59a900-e0c5-11eb-96f2-a56e149f8615.gif)

(Mac OSX, Firefox; Safari is similar but a bit more annoying because it keeps trying to select elements when I don't click or drag quite right)

Edit: smaller gif

**Reply by Jermolene** [reply]
Bravo @BurningTreeC, that's very promising. I couldn't find the implementation tiddlers, perhaps it's worth now either making a draft PR or posting an otherwise empty wiki. Many thanks!

**Reply by BurningTreeC** [reply]
Oh, I've posted the wrong .zip file 😁 

Here's the right one with the implementation tiddlers open in the story river:

[tiddlywiki.zip](https://github.com/Jermolene/TiddlyWiki5/files/6794825/tiddlywiki.zip)

**Comment by pmario** [comment]
I think, it would be a nice addition .. but with a much more subtle slider handle ;)

**Reply by BurningTreeC** [reply]
Thanks @pmario , yes the slider handle should either be transparent or a suitable svg positioned vertically centered

**Comment by BurningTreeC** [comment]
This is the latest actualized version:

[tiddlywiki.zip](https://github.com/Jermolene/TiddlyWiki5/files/6794843/tiddlywiki.zip)

**Comment by Alzacon** [comment]
Awesome idea.
